### ISSUE TITLE
Can't load new models

### ISSUE THREAD

**Issue by svenkevs** [huggingface/swift-coreml-diffusers#68]
Downloading the new models gives a 404 if you build the app from git. Xcode gives the following error:

```
Starting download of https://huggingface.co/apple/coreml-stable-diffusion-xl-base/resolve/main/coreml-stable-diffusion-xl-base_split_einsum_compiled.zip
HTTP response status code: 404
Could not load model, error: unreadableArchive
```

If such a new model that's not yet online is selected, you can't use the 'reveal in finder' option.

**Comment by pcuenca** [comment]
Hi @svenkevs! In your original message, you mentioned the `split_einsum` version of the model, which is not available yet – only the `original` attention implementation has been converted. However, the UI lets you select it (indirectly, by choosing the Neural Engine compute hardware). I'll make a fix to make it clear that you can only select the available versions.

Regarding your comment about the placement of the files, are you seeing it in the normal version or the 4.5-bit one? It worked fine in my tests, but I may have missed something in the release!

**Comment by ZachNagengast** [comment]
I think it's just the file structure of the new zips on hugging face, heres how they are currently:
<img width="500" alt="image" src="https://github.com/huggingface/swift-coreml-diffusers/assets/1981179/930bdbbe-cc37-4913-adb3-6e5efb6512c4">
And here's what the app is looking for:
<img width="487" alt="image" src="https://github.com/huggingface/swift-coreml-diffusers/assets/1981179/fe540fa4-ac66-4b2a-b1c3-81f39eb10837">

I'm not sure why it would work in some cases but not others, but that's the change I had to do (for the 4.5 bit model)

**Comment by pcuenca** [comment]
Interesting, that was my mistake then! I'm not sure why it worked for me when I tested, but I'll fix it. Thanks @ZachNagengast!

**Comment by pcuenca** [comment]
I replaced the zip file in the repo with the correct version (I hope 😅). My apologies for the trouble!

I'll open a new issue to prevent selection of the `split_einsum` version when it's not available.

Thanks for reporting!

**Comment by svenkevs** [comment]
Thanks for solving it so quickly! Though really it wasn't a big issue, the error reporting system was good enough to figure it all out quickly :)

**Comment by beezbo** [comment]
I also had this issue and I cloned the project yesterday.

**Comment by pcuenca** [comment]
@beezbo There are two issues here:

- The file structure of the palettized SDXL model was wrong. This is fixed.
- The SDXL models are only available for GPU, but you are allowed to select the Neural Engine. If you do, the contents of the models folder may be removed. This is yet to be fixed.

TL;DR: please, only use `GPU` with the SDXL models for now.

**Comment by jrittvo** [comment]
For the benefit of someone (me) trying to maintain a very basic understanding of all this, could you possibly explain in a sentence or two what the challenge is to making SDXL models work with the Neural Engine?  I have seen it said that it is because the basic architecture of the NE Unet pipeline is inflexible, or otherwise hard to adapt to new model specifications.  Is that the gist of the issue?

**Comment by GimmeSomeJazz** [comment]
Hi everyone!
I was wondering if it is possible to load new models (e.g. SDXL) directly in the compiled MacOS App (just modifying some conf file and adding the models in the /Users/USERNAME/Library/Containers/com.huggingface.Diffusers/Data/Library/Application Support/hf-diffusion-models directory.
Does this make sense?
Thanks!
M.

**Comment by dolmere** [comment]
@GimmeSomeJazz not at the moment. Keep an eye on PR 61 https://github.com/huggingface/swift-coreml-diffusers/pull/61

**Comment by doabigcheese** [comment]
@ghost any update on this? the PR 61 is closed but custom models still not integrated...
